test/realm_logger_test.dart

```diff
@@ -0,0 +1,142 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+// ignore_for_file: unused_local_variable, avoid_relative_lib_imports
+
+import 'dart:async';
+import 'dart:isolate';
+import 'package:logging/logging.dart';
+import 'package:test/test.dart' hide test, throws;
+import '../lib/realm.dart';
+import 'test.dart';
+
+Future<void> main([List<String>? args]) async {
+  await setupTests(args, loggerTest: true);
+  baasTest('Realm loggers log messages in all the isolates', (appConfiguration) async {
```

Comment:
Can we do a test for checking an expected message from Core is really logged by the logger. Also for different expected levels

Reply:
Let's discuss this.

Reply:
It appeared that the tests can not be isolated if we run a new isolate per test, because the first logger is alway replaced in the main isolate by caling `setupTests`.